crawler.py

```diff
@@ -81,21 +82,64 @@
 LOG_ERR = 'ERROR: "{}" - {}'
 
 
+class SearchConfig:
+    global config
+    config = {}
+
+    def __init__(self):
+        with open("config.yaml", "r") as config_file:
+            for data in yaml.load_all(config_file):
+                if "search" in data:
+                    yaml_data = data["search"]
+                    config["rooms_min"] = yaml_data["rooms_min"]
+                    config["rooms_max"] = yaml_data["rooms_max"]
+                    config["area_min"] = yaml_data["area_min"]
+                    config["area_max"] = yaml_data["area_max"]
+                    config["floor_min"] = yaml_data["floor_min"]
+                    config["floor_max"] = yaml_data["floor_max"]
+                    config["rent_max"] = yaml_data["rent_max"]
+                    config["wbs"] = "true" if yaml_data["wbs"] else "false"
+                    config["no_1st_floor"] = True if config["floor_min"] > 1 else False
+                    config["degewo_districts"] = "%2C+".join(
+                        yaml_data["degewo_districts"]
+                    )
+                    config["gewobag_districts"] = "&bezirke%5B%5D=".join(
+                        yaml_data["gewobag_districts"]
+                    )
+                if "site-blocklist" in data:
+                    config["site-blocklist"] = data["site-blocklist"]
+            config_file.close()
+
+    def get_config(self):
+        return config
+
+
 class Site:
     """
     A website to be searched for new flats. Takes a *config* dict, which should include
     most of the fields specified in `sites.py`.
     """
 
-    def __init__(self, config={}):
+    def __init__(self, config={}, search_config=None):
```

Comment:
If it gains a `search_config` argument
  a) the `config` argument should be renamed to `site_config`
  b) the new argument should be mentioned in the docstring above.

Reply:
Good point, the "config" member of the "Site" class should stay unchanged though, since it implies being the "site_config". What do you think?